mesonbuild/ast/interpreter.py

```diff
@@ -192,6 +192,16 @@ def evaluate_ternary(self, node: TernaryNode) -> None:
         self.evaluate_statement(node.trueblock)
         self.evaluate_statement(node.falseblock)
 
+    def evaluate_dictstatement(self, node: mparser.DictNode) -> T.Dict[str, T.Any]:
```

Comment:
This should return `TYPE_nkwargs`, not `T.Dict[str, T.Any]`.

Reply:
I don't understand why, but I'll oblige. :)

Reply:
It's because that is also the exact return type from the `InterpreterBase` and otherwise `mypy` would be very unhappy once #7657 is merged :)